Sources/AsyncHTTPClient/HTTPClient.swift

```diff
@@ -899,6 +900,13 @@ extension ChannelPipeline {
         try sync.addHandler(decoder)
         try sync.addHandler(handler)
     }
+    
+    func syncAddSOCKSProxyHandler(host: String, port: Int) throws {
+        let address = try SocketAddress(ipAddress: host, port: port)
+        let handler = SOCKSClientHandler(targetAddress: .address(address))
+        let sync = self.syncOperations
+        try sync.addHandler(handler)
+    }
```

Comment:
Is this something for NIO extras?

Reply:
We don't normally offer those extensions for protocols comprised of a single handler though.

Reply:
This is also slightly different to the pattern I think you mean, here I've just replicated what is done for the HTTP proxy above.